src/App.tsx

```diff
@@ -1,18 +1,49 @@
+  { id: 'profile', title: 'Profile', content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia enim doloremque repellendus laudantium. At quibusdam illo pariatur vitae quidem labore omnis dolorum quaerat. Facilis beatae dolor expedita corrupti distinctio necessitatibus debitis qui dolores velit, dolore sint vero, at cum porro dignissimos impedit ipsum mollitia assumenda asperiores cumque non sit doloremque. Eveniet dolorem totam commodi quasi, eum recusandae velit molestias voluptate accusamus vel esse sed debitis omnis eius deserunt ut cum earum architecto nisi ratione quam dignissimos, tenetur harum? Voluptate ipsum suscipit facilis minima fuga ab error iste deleniti dolor esse laudantium facere sed earum, numquam eaque dolorem doloremque aperiam vitae. Similique illum amet eos corporis cupiditate fugiat atque dolore, sapiente id. Pariatur maiores illum voluptatem, commodi voluptatum voluptatibus quibusdam sed doloremque dignissimos facilis veniam ipsum cumque molestiae, facere sit magnam enim consequuntur, itaque mollitia sunt delectus placeat eveniet quidem? Aliquam illo, explicabo quo quos consequatur ipsa fugit impedit laudantium, commodi magnam delectus!' },
+  { id: 'contact', title: 'Contact', content: 'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Repudiandae nesciunt eaque laboriosam explicabo, facilis quae adipisci unde. Sit veritatis blanditiis atque, est odit rerum praesentium molestias laborum doloribus, iusto nam! Ab aliquid quos odio velit nobis autem quaerat cum deleniti commodi voluptatibus, optio eum sed quasi dolorum necessitatibus cupiditate labore ipsum. Obcaecati, sunt!' },
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-    <div>{tabs.length}</div>
-  </div>
-);
+type State = {
+  tabs: Tab[],
+  selectedTabId: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    tabs: tabsFromServer,
+    selectedTabId: tabsFromServer[0].id,
+  };
+
+  select = (id: string) => {
+    this.setState(() => ({
+      selectedTabId: id,
+    }));
```

Comment:
why do you need functional setState here?
```suggestion
    this.setState({ selectedTabId: id });
```

Reply:
fixed